langchain/vectorstores/azuresearch.py

```diff
@@ -229,32 +213,45 @@ def add_texts(
         else:
             raise Exception(response)
 
-    def similarity_search(
-        self, query: str, k: int = 4, **kwargs: Any
+
+    def similarity_search(self, query: str, k: int = 50, **kwargs: Any):
+        search_type = kwargs.get("search_type", self.search_type)
+        if search_type == "similarity":
+            docs = self.vector_search(query, k=k)
+        elif search_type == "hybrid":
+            docs = self.hybrid_search(query, k=k)
+        elif search_type == "semantic_hybrid":
+            docs = self.semantic_hybrid_search(query, k=k)
+        else:
+            raise ValueError(f"search_type of {search_type} not allowed.")
+        return docs
+
+    def vector_search(
+        self, query: str, k: int = 50, **kwargs: Any
     ) -> List[Document]:
         """
         Returns the most similar indexed documents to the query text.
 
         Args:
             query (str): The query text for which to find similar documents.
-            k (int): The number of documents to return. Default is 4.
+            k (int): The number of documents to return. Default is 50.
```

Comment:
Why the change to a default of 50 documents?  This seems like it will almost always be way too many results to feed into a LLM without exceeding the max number of tokens.

Reply:
+1. @ruoccofabrizio when I suggested that we change k to 50, I meant the "k" parameter to vector queries, and only when using hybrid search. Separately, the number we pass to the "top" query parameter should still be small in most cases (at least by default) since that's included into prompts.

Reply:
fully agree, good catch. It was not meant to be modified outside of semantic search. Fixed ;)